core/domain/story_services.py

```diff
@@ -415,13 +417,26 @@ def _save_story(committer_id, story, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (story_model.version, story.version))
 
+    topic = topic_services.get_topic_by_id(story.belongs_to_topic, strict=False)
+    if topic is None:
+        raise utils.ValidationError(
+            'Expected story to only belong to a valid topic, but found an '
+            'topic with ID: %s' % story.belongs_to_topic)
+
+    if story.id not in topic.canonical_story_ids + topic.additional_story_ids:
+        raise Exception(
+            'Expected story to belongs to the topic %s, but it is '
+            'neither a part of canonical story or the addition story of '
```

Comment:
'the canonical stories or the additional stories'

Reply:
Done!